source/module_base/vector3.h

```diff
@@ -36,6 +36,9 @@ template <class T> class Vector3
     Vector3(const Vector3<T> &v) : x(v.x), y(v.y), z(v.z){}; // Peize Lin add 2018-07-16   
     explicit Vector3(const std::array<T,3> &v) :x(v[0]), y(v[1]), z(v[2]){}
 
+    template <typename U>
```

Comment:
what's the reason to add this new function?

Reply:
This is an explicit constructor that allows a vector3 of int type to be used to generate a vector3 of double type. this function is used in https://github.com/deepmodeling/abacus-develop/pull/5568/files#diff-4de1e02af83e394ccb0f0c001bc62d08bc4df0ef3ed6f7ecef5b2f29d4c67ddbL94-R91